File: webapp/app/_types/resume-processor.ts
export type VectorScore = {
  jobId: string;
  score: number;
};

export type CommonWords = {
  jobId: string;
  text: string;
};

export type SuggestionChanges = {
  changeFrom: string;
  changeTo: string;
};

export type Suggestion = {
  jobId: string;
  changes: SuggestionChanges[];
};

export type ResumeProcessorResponse = {
  vectorScoresSet: VectorScore[];
  commonWordsSet: CommonWords[];
  suggestionsSet: Suggestion[];
};

File: webapp/backend/schemas/resume_processor.py
from typing import List, Optional
from fastapi import File, Form, UploadFile
from pydantic import BaseModel


class Job(BaseModel):
    id: str
    link: Optional[str] = None
    description: Optional[str] = None


class ResumeProcessorRequest(BaseModel):
    resume: UploadFile
    jobs: str

    @classmethod
    def as_form(cls, resume: UploadFile = File(...), jobs: str = Form(...)):
        return cls(resume=resume, jobs=jobs)


class VectorScore(BaseModel):
    jobId: str
    score: int


class CommonWord(BaseModel):
    jobId: str
    text: str


class Changes(BaseModel):
    changeFrom: str
    changeTo: str


class Suggestion(BaseModel):
    jobId: str
    changes: List[Changes]


class ResumeProcessorResponse(BaseModel):
    vectorScoresSet: List[VectorScore]
    commonWordsSet: List[CommonWord]
    suggestionsSet: List[Suggestion]
